### PR TITLE
Feature/countdown component

### PR DESCRIPTION
**Pull request by ZanBin**
## Check List
- [ ] Your code builds clean without any errors or warnings.
- [ ] Check coding convention.
- [ ] Updated Storybook components and stories to reflect any changes made to the UI.
- [ ] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Screenshot changes introduced" width="250" />
OR
Video link

### CODE REVIEW

File: src/components/CountDown/CountDown.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  const timeEnd = useRef(new Date(new Date().getTime() + initialSeconds * 1000).getTime())
+  const [appStateVisible, setAppStateVisible] = useState(appState.current)
+
+  /*
+    check app state is change
+  */
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', nextAppState => {
+      if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
+        console.log('App has come to the foreground!')
```
Comment (loc-nguyenthien) [inline src/components/CountDown/CountDown.tsx:52]: Do we do any thing in here?

File: src/components/CountDown/CountDown.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  /*
+  loop countdown
+  */
+  loop?: boolean
+  /*
+  format countdown
+  */
+  format?: 'MM:SS' | 'HH:MM:SS' | 'DD:HH:MM:SS'
+}
+const intervalTimeBySecond = 1000
```
Comment (ledutu2) [inline src/components/CountDown/CountDown.tsx:31]: I think you should pass intervalTimeBySecond as a prop with default = 1000

File: src/components/CountDown/CountDown.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            onFinish()
+          }
+        }
+      }
+    }, intervalTimeBySecond)
+    return () => {
+      clearInterval(interval)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [seconds, appStateVisible])
```
Comment (loc-nguyenthien) [inline src/components/CountDown/CountDown.tsx:84]: This `setInterval` is change every second we setSeconds, maybe we should just use `setTimeout`, that way we won't need to clearInterval

File: src/components/CountDown/CountDown.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const minute = Math.floor(seconds / 60) % 60 >= 0 ? Math.floor(seconds / 60) % 60 : 0
+    const textMinute = minute >= 10 ? `${minute}p` : `0${minute}p`
+    /*
+    caculate seconds
+    */
+    const _seconds = Math.round(seconds % 60) >= 0 ? Math.round(seconds % 60) : 0
```
Comment (loc-nguyenthien) [inline src/components/CountDown/CountDown.tsx:95]: This `seconds` is always a positive number, right? So we can just
```suggestion
    const _seconds = Math.round(seconds % 60)
```

File: src/components/CountDown/CountDown.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      clearInterval(interval)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [seconds, appStateVisible])
+
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const minute = Math.floor(seconds / 60) % 60 >= 0 ? Math.floor(seconds / 60) % 60 : 0
```
Comment (loc-nguyenthien) [inline src/components/CountDown/CountDown.tsx:90]: Check my below comment

File: src/components/CountDown/CountDown.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    const textSecond = _seconds >= 10 ? `${_seconds}s` : `0${_seconds}s`
+
+    let textDay = null
+    let textHour = null
+    /*
+    caculate day
+    */
+    if (format.includes('DD')) {
+      const caculate = 60 * 60 * 24
+      const day = Math.floor(seconds / caculate) >= 0 ? Math.floor(seconds / caculate) : 0
```
Comment (loc-nguyenthien) [inline src/components/CountDown/CountDown.tsx:105]: Check my above comment

File: src/components/CountDown/CountDown.tsx [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+      textDay = day >= 10 ? `${day}d` : `0${day}d`
+    }
+    /*
+    caculate hours
+    */
+    if (format.includes('HH')) {
+      const caculate = 60 * 60
+      const hour =
+        Math.floor(seconds / caculate) % caculate >= 0 ? Math.floor(seconds / caculate) % caculate : 0
```
Comment (loc-nguyenthien) [inline src/components/CountDown/CountDown.tsx:114]: Please recheck this calculation, I don't think we need `% caculate`, and check my above comment

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: src/components/CountDown/CountDown.tsx (hunk 1 above)
Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx:52]: in every device we have 2 status for device when we using app
1: background
2: active
so when the device in background status the setTimeout/setInteral are't working
so your countdown event will wrong
that why i check the app status to make sure it always correct

File: src/components/CountDown/CountDown.tsx (hunk 4 above)
Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx:95]: Check my below comment
when the app in background and your countdown and the time large than the time end it will be become negative number

File: src/components/CountDown/CountDown.tsx (hunk 3 above)
Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx:84]: i will check it

File: src/components/CountDown/CountDown.tsx (hunk 2 above)
Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx:31]: sure, i will change it

File: src/components/CountDown/CountDown.tsx (hunk 7 above)
Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx]: sure
i'm checking it

File: src/components/CountDown/CountDown.tsx [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+          timeEnd.current = new Date(new Date().getTime() + initialSeconds * 1000).getTime()
+          setSeconds(initialSeconds)
+        } else {
+          clearTimeout(timeout)
+          if (onFinish) {
+            onFinish()
+          }
+        }
+      }
+    }, intervalTimeBySecond)
```
Comment (honghoangsts) [inline src/components/CountDown/CountDown.tsx:80]: the time shouldn't === 1000ms. because `setTimeout` not run after exactly 1000ms. so it will make the time update not correct.

File: src/components/CountDown/CountDown.tsx [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    return () => {
+      subscription.remove()
+    }
+  }, [])
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (seconds >= 0) {
+        const uff = new Date(timeEnd.current - new Date().getTime()).getTime()
+        setSeconds(uff / 1000)
```
Comment (honghoangsts) [inline src/components/CountDown/CountDown.tsx:67]: shouldn't have magic number

File: src/components/CountDown/CountDown.tsx [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const minute = Math.floor(seconds / 60) % 60 >= 0 ? Math.floor(seconds / 60) % 60 : 0
+    const textMinute = minute >= 10 ? `${minute}p` : `0${minute}p`
+    /*
+    caculate seconds
+    */
+    const _seconds = Math.round(seconds % 60) >= 0 ? Math.round(seconds % 60) : 0
```
Comment (honghoangsts) [inline src/components/CountDown/CountDown.tsx:96]: we not user `underscore` when declare a variable.

File: src/components/CountDown/CountDown.tsx [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    */
+    if (format.includes('DD')) {
+      const caculate = 60 * 60 * 24
+      const day = Math.floor(seconds / caculate) >= 0 ? Math.floor(seconds / caculate) : 0
+      textDay = day >= 10 ? `${day}d` : `0${day}d`
+    }
+    /*
+    caculate hours
+    */
+    if (format.includes('HH')) {
```
Comment (honghoangsts) [inline src/components/CountDown/CountDown.tsx:112]: don't use magic string

File: src/components/CountDown/CountDown.tsx [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+      const caculate = 60 * 60
+      const hour = Math.floor(seconds / caculate) >= 0 ? Math.floor(seconds / caculate) % 24 : 0
+      textHour = hour >= 10 ? `${hour}h` : `0${hour}h`
+    }
+    return (
+      <Text style={[textStyle]}>
```
Comment (honghoangsts) [inline src/components/CountDown/CountDown.tsx:118]: we should allow custom the way render time. refer to shoppe. the count down contain in the box

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/components/CountDown/CountDown.tsx [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+        {textDay && `${textDay}:`}
+        {textHour && `${textHour}:`}
+        {textMinute}:{textSecond}
+      </Text>
+    )
+  }
+
+  return <Container style={containerStyle}>{seconds === 0 ? null : renderTimer()}</Container>
```
Comment (hangnguyensaigontech) [inline src/components/CountDown/CountDown.tsx:126]:
```suggestion
  return <Container style={containerStyle}>{!!seconds && renderTimer()}</Container>
```

File: src/components/CountDown/CountDown.tsx [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+  }
+
+  return renderTimer()
+}
+
+export default CountDown
+const Container = styled.View({
+  flexDirection: 'row',
+})
+const Text = styled.Text(() => ({
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:175]: The `Text` should be use from our `rn-base-component`

File: src/components/CountDown/CountDown.tsx [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+  /*
+  format countdown
+  */
+  format?: FormatTime
+  /*
+  default count time after milisecond
+  */
+  intervalTimeBySecond?: number
+}
+const CountDown: React.FunctionComponent<CountDownProps> = ({
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:45]:
```suggestion
export const CountDown: React.FunctionComponent<CountDownProps> = ({
```

File: src/components/CountDown/CountDown.tsx [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+        {renderColons()}
+        {textSecond}
+      </Container>
+    )
+  }
+
+  return renderTimer()
+}
+
+export default CountDown
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:171]: remove this line

File: src/components/CountDown/CountDown.tsx [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+    )
+  }
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const minute = Math.floor(seconds / 60) % 60 >= 0 ? Math.floor(seconds / 60) % 60 : 0
+    const textMinute = (
+      <Container style={elementStyle}>
+        <Text style={[textStyle]}>{minute >= 10 ? `${minute}p` : `0${minute}p`}</Text>
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:114]: Please remove `p`

File: src/components/CountDown/CountDown.tsx [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+      </Container>
+    )
+    /*
+    caculate seconds
+    */
+    const second = Math.round(seconds % 60) >= 0 ? Math.round(seconds % 60) : 0
+    const textSecond = (
+      <Container style={elementStyle}>
+        <Text style={[textStyle]}>{second >= 10 ? `${second}s` : `0${second}s`}</Text>
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:123]: Please remove `s`

File: src/components/CountDown/CountDown.tsx [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+    const formatHour = 'HH'
+    /*
+    caculate day
+    */
+    if (format.includes(formatDate)) {
+      const caculate = 60 * 60 * 24
+      const day = Math.floor(seconds / caculate) >= 0 ? Math.floor(seconds / caculate) : 0
+      textDay = (
+        <Container style={elementStyle}>
+          <Text style={[textStyle]}>{day >= 10 ? `${day}d` : `0${day}d`}</Text>
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:138]: Please remove `d`

File: src/components/CountDown/CountDown.tsx [hunk 20]
```diff
@@ -0,0 +1,177 @@
...
+    }
+    /*
+    caculate hours
+    */
+    if (format.includes(formatHour)) {
+      const caculate = 60 * 60
+      const hour = Math.floor(seconds / caculate) >= 0 ? Math.floor(seconds / caculate) % 24 : 0
+      textHour = (
+        <Container style={elementStyle}>
+          <Text style={[textStyle]}>{hour >= 10 ? `${hour}h` : `0${hour}h`}</Text>
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:150]: Please remove `h`

File: src/components/CountDown/CountDown.tsx [hunk 21]
```diff
@@ -0,0 +1,177 @@
+import React, {useEffect, useRef, useState} from 'react'
+import {AppState, type StyleProp, type ViewStyle, type TextStyle} from 'react-native'
+import styled from 'styled-components/native'
+
+type FormatTime = 'MM:SS' | 'HH:MM:SS' | 'DD:HH:MM:SS'
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:5]:
```suggestion
const FormatTime = {
mmss: 'mm:ss',
hhmmss: 'HH:mm:ss',
ddhhmmss: 'dd:HH:mm:ss',
} as const
```

File: src/components/CountDown/CountDown.tsx [hunk 22]
```diff
@@ -0,0 +1,177 @@
...
+  */
+  colonsStyle?: StyleProp<TextStyle>
+  /*
+  loop countdown
+  */
+  loop?: boolean
+  /*
+  format countdown
+  */
+  format?: FormatTime
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:39]:
```suggestion
  format?: keyof typeof FormatTime
```

File: src/components/CountDown/CountDown.tsx [hunk 23]
```diff
@@ -0,0 +1,177 @@
...
+  */
+  intervalTimeBySecond?: number
+}
+const CountDown: React.FunctionComponent<CountDownProps> = ({
+  initialSeconds = 300,
+  containerStyle,
+  onFinish,
+  textStyle,
+  loop,
+  format = 'MM:SS',
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:51]:
```suggestion
  format = FormatTime.mmss,
```

File: src/components/CountDown/CountDown.tsx [hunk 24]
```diff
@@ -0,0 +1,177 @@
...
+  intervalTimeBySecond = 1000,
+  elementStyle,
+  colonsStyle,
+}) => {
+  const [seconds, setSeconds] = useState(initialSeconds)
+
+  const appState = useRef(AppState.currentState)
+  const timeEnd = useRef(new Date().getTime() + initialSeconds * 1000)
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:59]: create constants

File: src/components/CountDown/CountDown.tsx [hunk 25]
```diff
@@ -0,0 +1,177 @@
...
+      subscription.remove()
+    }
+  }, [])
+
+  useEffect(() => {
+    const timeLoop = intervalTimeBySecond - new Date().getMilliseconds()
+    const timeout = setTimeout(() => {
+      if (seconds >= 0) {
+        const uff = timeEnd.current - new Date().getTime()
+        setSeconds(uff / 1000)
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:80]: create constants for 1000

File: src/components/CountDown/CountDown.tsx [hunk 26]
```diff
@@ -0,0 +1,177 @@
...
+      <Container>
+        <Text style={[textStyle, colonsStyle]}>:</Text>
+      </Container>
+    )
+  }
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const minute = Math.floor(seconds / 60) % 60 >= 0 ? Math.floor(seconds / 60) % 60 : 0
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:111]: create constants for 60

File: src/components/CountDown/CountDown.tsx [hunk 27]
```diff
@@ -0,0 +1,177 @@
...
+    const textMinute = (
+      <Container style={elementStyle}>
+        <Text style={[textStyle]}>{minute >= 10 ? `${minute}p` : `0${minute}p`}</Text>
+      </Container>
+    )
+    /*
+    caculate seconds
+    */
+    const second = Math.round(seconds % 60) >= 0 ? Math.round(seconds % 60) : 0
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:120]: create constants for 60

File: src/components/CountDown/CountDown.tsx [hunk 28]
```diff
@@ -0,0 +1,177 @@
...
+    )
+    let textDay = null
+    let textHour = null
+    const formatDate = 'DD'
+    const formatHour = 'HH'
+    /*
+    caculate day
+    */
+    if (format.includes(formatDate)) {
+      const caculate = 60 * 60 * 24
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:134]: create constants for 60, 24

File: src/components/CountDown/CountDown.tsx [hunk 29]
```diff
@@ -0,0 +1,177 @@
...
+    */
+    const second = Math.round(seconds % 60) >= 0 ? Math.round(seconds % 60) : 0
+    const textSecond = (
+      <Container style={elementStyle}>
+        <Text style={[textStyle]}>{second >= 10 ? `${second}s` : `0${second}s`}</Text>
+      </Container>
+    )
+    let textDay = null
+    let textHour = null
+    const formatDate = 'DD'
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:128]:
```suggestion
    const formatDate = 'dd'
```

File: src/components/CountDown/CountDown.tsx [hunk 30]
```diff
@@ -0,0 +1,177 @@
...
+    const formatHour = 'HH'
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:129]:
```suggestion
    const formatHour = 'hh'
```

File: src/components/CountDown/CountDown.tsx [hunk 31]
```diff
@@ -0,0 +1,177 @@
...
+        <Container style={elementStyle}>
+          <Text style={[textStyle]}>{day >= 10 ? `${day}d` : `0${day}d`}</Text>
+        </Container>
+      )
+    }
+    /*
+    caculate hours
+    */
+    if (format.includes(formatHour)) {
+      const caculate = 60 * 60
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:146]: create constants for 60

File: src/components/CountDown/CountDown.tsx [hunk 32]
```diff
@@ -0,0 +1,177 @@
...
+      const hour = Math.floor(seconds / caculate) >= 0 ? Math.floor(seconds / caculate) % 24 : 0
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:147]: create constants for 24

File: src/components/CountDown/CountDown.tsx [hunk 33]
```diff
@@ -0,0 +1,177 @@
+import React, {useEffect, useRef, useState} from 'react'
+import {AppState, type StyleProp, type ViewStyle, type TextStyle} from 'react-native'
+import styled from 'styled-components/native'
+
+type FormatTime = 'MM:SS' | 'HH:MM:SS' | 'DD:HH:MM:SS'
+
+export type CountDownProps = {
```
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:7]: I think we need a prop `is24hour`

File: src/components/CountDown/CountDown.tsx (hunk 22 above)
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:39]:
```suggestion
  format?: keyof typeof FormatTime
```

File: src/components/CountDown/CountDown.tsx (hunk 23 above)
Comment (baonguyenhsts) [inline src/components/CountDown/CountDown.tsx:51]:
```suggestion
  format = FormatTime.mmss,
```

File: src/components/CountDown/CountDown.tsx [hunk 34]
```diff
@@ -0,0 +1,181 @@
...
+import {AppState, type StyleProp, type ViewStyle, type TextStyle} from 'react-native'
+import styled from 'styled-components/native'
+import {Text} from 'rn-base-component'
+export const FormatTime = {
+  mmss: 'mm:ss',
+  hhmmss: 'HH:mm:ss',
+  ddhhmmss: 'DD:HH:mm:ss',
+} as const
+
+export type CountDownProps = {
```
Comment (chungnguyenhoang) [inline src/components/CountDown/CountDown.tsx:11]: please extend TextProps if you using a Text from rn-base-component

File: src/components/CountDown/CountDown.tsx [hunk 35]
```diff
@@ -0,0 +1,181 @@
...
+      </Container>
+    )
+  }
+
+  return renderTimer()
+}
+
+const Container = styled.View({
+  flexDirection: 'row',
+})
```
Comment (chungnguyenhoang) [inline src/components/CountDown/CountDown.tsx:181]: create a const TextBase = styled.Text<TextProps>({theme})...

view some different component to make right

File: src/components/CountDown/CountDown.tsx [hunk 36]
```diff
@@ -0,0 +1,181 @@
...
+        {renderColons()}
+        {textSecond}
+      </Container>
+    )
+  }
+
+  return renderTimer()
+}
+
+const Container = styled.View({
```
Comment (chungnguyenhoang) [inline src/components/CountDown/CountDown.tsx:179]: please create a component theme same like ButtonComponent

Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx:179]: can you recommend some components

File: src/theme/components/Countdown/Countdown.ts [hunk 37]
```diff
@@ -0,0 +1,12 @@
+import base from '../../base'
+import {metrics} from '../../../helpers'
+import type {CountDownProps} from '../../../components/CountDown/CountDown'
+
+export type CountDownThemeProps = {} & Pick<CountDownProps, 'textStyle'>
```
Comment (baonguyenhsts) [inline src/theme/components/Countdown/Countdown.ts:5]:
```suggestion
export type CountDownThemeProps = Pick<CountDownProps, 'textStyle'>
```

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: src/components/CountDown/CountDown.tsx [hunk 38]
```diff
@@ -34,29 +38,29 @@ export type CountDownProps = {
...
+  format = 'mmss',
   intervalTimeBySecond = 1000,
   elementStyle,
   colonsStyle,
 }) => {
   const [seconds, setSeconds] = useState(initialSeconds)
-
   const appState = useRef(AppState.currentState)
-  const timeEnd = useRef(new Date().getTime() + initialSeconds * 1000)
+  const milisecond = 1000
```
[the remark this answers is not in the document]
Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx:66]: because we have initialSeconds to run the countdown
so i think it will hard for user if they must check the time end

[the remark this answers is not in the document]
Comment (baonguyenhsts) [reply src/components/CountDown/CountDown.tsx:66]: What I mean is that the variable `milisecond` will be re-created if the component render so we need to move it outside like constants

[the remark this answers is not in the document]
Comment (ZanBin) [reply src/components/CountDown/CountDown.tsx:66]: yes
i will move all const outside the component

File: src/components/CountDown/CountDown.tsx [hunk 39]
```diff
@@ -0,0 +1,191 @@
...
+    <Container>
+      <Label style={[CountDownTheme.textStyle, textStyle, colonsStyle]}>:</Label>
+    </Container>
+  )
+
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const resultCaculatMinute = Math.floor(seconds / numberSecondPerMinute) % numberSecondPerMinute
```
Comment (hangnguyensaigontech) [inline src/components/CountDown/CountDown.tsx:126]: wrong typo
```suggestion
    const resultCaculateMinute = Math.floor(seconds / numberSecondPerMinute) % numberSecondPerMinute
```

File: src/components/CountDown/CountDown.tsx [hunk 40]
```diff
@@ -0,0 +1,191 @@
...
+    const minute = resultCaculatMinute >= 0 ? resultCaculatMinute : 0
```
Comment (hangnguyensaigontech) [inline src/components/CountDown/CountDown.tsx:127]: why it need to be `>= 0` ? shouldn't it be only `> 0`

if resultCaculateMinute = 0, it will be same as the value return from else condition

File: src/components/CountDown/CountDown.tsx [hunk 41]
```diff
@@ -0,0 +1,191 @@
...
+    const textMinute = (
+      <Container style={elementStyle}>
+        <Label style={[CountDownTheme.textStyle, textStyle]}>{minute.toString().padStart(2, '0')}</Label>
+      </Container>
+    )
+    /*
+    caculate seconds
+    */
+    const resultCaculatSecond = Math.round(seconds % numberSecondPerMinute)
+    const second = resultCaculatSecond >= 0 ? resultCaculatSecond : 0
```
Comment (hangnguyensaigontech) [inline src/components/CountDown/CountDown.tsx:137]: same with `const minute`

File: src/components/CountDown/CountDown.tsx [hunk 42]
```diff
@@ -0,0 +1,191 @@
...
+    )
+    let textDay = null
+    let textHour = null
+
+    /*
+    caculate day
+    */
+    if (FormatTime.ddhhmmss === FormatTime[format]) {
+      const resultCaculatDay = Math.floor(seconds / secondPerDay)
+      const day = resultCaculatDay >= 0 ? resultCaculatDay : 0
```
Comment (hangnguyensaigontech) [inline src/components/CountDown/CountDown.tsx:151]: also here

File: src/components/CountDown/CountDown.tsx [hunk 43]
```diff
@@ -0,0 +1,194 @@
...
+      textHour = (
+        <Container style={elementStyle}>
+          <Label style={[CountDownTheme.textStyle, textStyle]}>{hour.toString().padStart(2, '0')}</Label>
+        </Container>
+      )
+    }
+
+    return (
+      <Container style={[CountDownTheme.containerStyle, containerStyle]}>
+        {textDay && textDay}
```
Comment (hangnguyensaigontech) [inline src/components/CountDown/CountDown.tsx:173]: should it convert to Boolean, because textDay has case = null, so if it return null it will return error string outside Text component
